samples/matter/common/src/bt_nus_service.cpp

```diff
@@ -60,41 +60,50 @@ bool NUSService::Init(uint8_t priority, uint16_t minInterval, uint16_t maxInterv
 
 	mAdvertisingRequest.onStarted = [](int rc) {
 		if (rc == 0) {
-			LOG_INF("NUS BLE advertising started");
+			GetNUSService().mIsStarted = true;
+			LOG_DBG("NUS BLE advertising started");
 		} else {
 			LOG_ERR("Failed to start NUS BLE advertising: %d", rc);
 		}
 	};
-	mAdvertisingRequest.onStopped = []() { LOG_DBG("NUS BLE advertising stopped"); };
+	mAdvertisingRequest.onStopped = []() {
+		bt_conn_auth_info_cb_unregister(&sConnAuthInfoCallbacks);
+		bt_conn_auth_cb_register(NULL);
+		bt_nus_init(NULL);
+
+		GetNUSService().mIsStarted = false;
+		LOG_DBG("NUS BLE advertising stopped");
+	};
 
 	settings_load();
 #if defined(CONFIG_BT_FIXED_PASSKEY)
 	VerifyOrReturnError(bt_passkey_set(CONFIG_CHIP_NUS_FIXED_PASSKEY) == 0, false);
 #endif
-	VerifyOrReturnError(bt_conn_auth_cb_register(&sConnAuthCallbacks) == 0, false);
-	VerifyOrReturnError(bt_conn_auth_info_cb_register(&sConnAuthInfoCallbacks) == 0, false);
-	VerifyOrReturnError(bt_nus_init(&sNusCallbacks) == 0, false);
 
 	return true;
 }
 
-void NUSService::StartServer()
+bool NUSService::StartServer()
 {
 	if (mIsStarted) {
 		LOG_WRN("NUS service was already started");
-		return;
+		return false;
 	}
 
+	VerifyOrReturnError(bt_conn_auth_cb_register(&sConnAuthCallbacks) == 0, false);
```

Comment:
nit: is there any difference between `VerifyOrReturnError` and `VerifyOrReturnValue`? Isn't the first one dedicated to CHIP_ERROR type?